modular_skyrat/modules/colony_fabricator/code/machines/arc_furnace.dm

```diff
@@ -0,0 +1,194 @@
+		if("use")
+			smelt_it_up(user)
+
+/obj/machinery/arc_furnace/proc/eject_contents()
+	if(operating)
+		return
+
+	playsound(loc, 'sound/machines/click.ogg', 15, TRUE, -3)
+
+	if(!length(contents))
+		return
+
+	var/atom/movable/thing_inside = contents[1]
+	thing_inside.forceMove(drop_location())
+	update_appearance()
+
+/obj/machinery/arc_furnace/proc/smelt_it_up(mob/user)
+	if(machine_stat & (NOPOWER|BROKEN))
+		balloon_alert(user, "button doesn't respond")
+		return
+	if(operating)
+		balloon_alert(user, "already smelting")
+		return
+
+	var/obj/item/stack/ore/ore_to_smelt = contents[1]
+	if(!istype(ore_to_smelt))
+		balloon_alert(user, "nothing to smelt")
+
+	operating = TRUE
+	/// How long the smelting is going to take based off the stack size
+	var/smelting_time = ore_to_smelt.amount * 1 SECONDS
+	loop(smelting_time)
+
+	soundloop.start()
+	set_light(l_range = 1.5)
+
+	update_appearance()
+
+/obj/machinery/arc_furnace/proc/loop(time)
+	if(machine_stat & (NOPOWER|BROKEN))
+		end_smelting()
+		return
+
+	if(!length(contents))
+		end_smelting()
+		return
+
+	if(time <= 0)
+		succeed_smelting()
+		return
+
+	time -= 1 SECONDS
+	use_power(active_power_usage)
+
+	var/turf/where_we_spawn_air = get_turf(src)
+	var/obj/item/stack/ore/ore_stack_to_check = contents[1]
+	switch(ore_stack_to_check.refined_type)
+		if(/obj/item/stack/sheet/mineral/silver)
+			where_we_spawn_air.atmos_spawn_air("n2=10;TEMP=1200")
+		if(/obj/item/stack/sheet/mineral/uranium)
+			where_we_spawn_air.atmos_spawn_air("co2=50;TEMP=1200")
+		if(/obj/item/stack/sheet/mineral/titanium)
+			where_we_spawn_air.atmos_spawn_air("n2=10;co2=10;TEMP=1200")
+		else
+			where_we_spawn_air.atmos_spawn_air("co2=20;TEMP=1200")
+
+	addtimer(CALLBACK(src, PROC_REF(loop), time), 1 SECONDS)
+
+/obj/machinery/arc_furnace/proc/succeed_smelting()
+	var/obj/item/stack/ore/ore_to_smelt = contents[1]
+	if(!istype(ore_to_smelt))
+		end_smelting()
+	/// Stack of product we just made and are going to fiddle with after spawning
+	var/obj/item/stack/smelting_product = new ore_to_smelt.refined_type(drop_location())
+	smelting_product.amount = ore_to_smelt.amount
+	smelting_product.update_appearance(UPDATE_ICON)
+	qdel(ore_to_smelt)
+	end_smelting()
+
+/obj/machinery/arc_furnace/proc/end_smelting()
+	operating = FALSE
+	soundloop.stop()
+	set_light(l_range = 0)
+	update_appearance()
+
+// Item for creating the arc furnace or carrying it around
+
+/obj/item/flatpacked_machine/arc_furnace
+	name = "\improper flatpacked arc furnace"
+	desc = "An arc furnace, a specialist machine that can rapidly smelt ores using, as the name implies, massive \
+		amounts of electricity. While not nearly as fast and efficient as other ore refining methods, none are anywhere \
+		near as portable as these are."
+	icon_state = "arc_furnace_folded"
+	type_to_deploy = /obj/machinery/arc_furnace
+	custom_materials = list(/datum/material/iron = SHEET_MATERIAL_AMOUNT * 7.5, /datum/material/glass = SHEET_MATERIAL_AMOUNT * 3)
```

Comment:
Might as well make it multiline, like the other lists of material datums, for consistency, yeah?